### PR TITLE
chore: add new docs section for Leaderboard

### PR DESCRIPTION
**Pull request by anitarua**
Addresses [#468](https://github.com/momentohq/dev-eco-issue-tracker/issues/468). Adds a new section for Leaderboard and an initial set of API reference pages.

### CODE REVIEW

File: docs/leaderboard/index.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+Once on the API key page, select the information that matches where your caches live:
+
+1. Cloud provider - AWS
+2. Region - us-west-2
+3. Key Type - Super User
+4. (Optional) Expiration date
+
+![Image showing the fields to create a new API key](@site/static/img/getting-started/select-provider-region-us-west-2.png)
+
+### Step 3: Create a Momento Leaderboard
```
Comment (cprice404) [inline docs/leaderboard/index.md]: nit: we go from step 1 to step 3 here

File: docs/leaderboard/index.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+title: Momento Leaderboard Documentation
+description: Learn about Momento's purpose-built Leaderboard API
+pagination_next: null
+---
+
+## What is Momento Leaderboard?
+
+Momento Leaderboard is a purpose-built leaderboard that supports tens of millions of items and rapid ingestion/querying/updates. While many databases approximate leaderboards via general purpose data structures like sorted set or range queries, Momento Leaderboard is a first-class, full-fledged service that you can quickly and easily integrate with your application. 
+
+## Getting Started with Momento Leaderboard
```
Comment (cprice404) [inline docs/leaderboard/index.md:23]: I think there is currently a server-side allowlist for this feature, so we will have to add a paragraph stating that users need to contact us to get added to that before they can use it. (and maybe show the error they will get if they try to use it w/o the allowlist?)

**Review by cprice404 — COMMENTED** [review]
@allenheltondev I think I'm fine with this other then the nits I left. This is adding a new section for leaderboards to the main nav though so we should make sure we're all aligned before it goes in.

File: docs/leaderboard/develop/api-reference/index.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+## Leaderboard Methods
+
+### Upsert elements
+
+Inserts elements if they do not already exist in the leaderboard. Updates elements if they do already exist in the leaderboard.
+
+| Name         | Type                | Description                                                    |
+|--------------|---------------------|----------------------------------------------------------------|
+| elements     | Map<number, number> | A Map or Record of elements (id,score pairs) to upsert.        |
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md:37]: Shouldn't this be `Map<string, number>`?

File: docs/leaderboard/develop/api-reference/index.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+* Success
+* Error
+
+See [response objects](./response-objects.md) for specific information.
+
+</details>
+
+<SdkExampleTabs snippetId={'API_LeaderboardUpsert'} />
+
+For large leaderboards, you will need to upsert in batches of 8192 elements.
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md]: This should be stated `of up to 8192 elements`

We should also explain why.

File: docs/leaderboard/develop/api-reference/index.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+| maxScore     | Optional[number]    | Exclusive upper bound for the score range. Defaults to `+inf`.    |
+| order        | Optional[Ascending / Descending]    | The order to fetch the elements in. Defaults to ascending, meaning 0 is the lowest-scoring rank.   |
+| offset       | Optional[number]    | The number of elements to skip before returning the first element. Defaults to 0. Note: this is not the score of the first element to return, but the number of elements of the result set to skip before returning the first element.    |
+| count        | Optional[number]    | The maximum number of elements to return. Defaults to 8192, which is the maximum that can be fetched per request.    |
+
+<details>
+  <summary>Method response object</summary>
+
+* Success
+    * values(): {`id`: number, `score`: number, `rank`: number}[]
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md:72]: Do we have an indicator there are more pages if the leaderboard contains more than 8192? Or if it contains more than the provided `count` property?

File: docs/leaderboard/develop/api-reference/index.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+* Error
+
+See [response objects](./response-objects.md) for specific information.
+
+</details>
+
+<SdkExampleTabs snippetId={'API_LeaderboardFetchByScore'} />
+
+For large leaderboards, you will need to fetch in batches of 8192 elements.
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md]: Instead of saying "you will need to fetch", we should say "you can fetch" since there's an actual limitation

File: docs/leaderboard/develop/api-reference/index.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+</details>
+
+<SdkExampleTabs snippetId={'API_LeaderboardFetchByRank'} />
+
+For large leaderboards, you will need to fetch in batches of 8192 elements.
+
+<SdkExampleTabs snippetId={'API_LeaderboardFetchByRankPagination'} />
+
+### Fetch elements' rank
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md]: This does not look like the correct name for this operation. Maybe `Fetch elements by id`?

File: docs/leaderboard/develop/api-reference/index.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+
+<SdkExampleTabs snippetId={'API_LeaderboardFetchByRank'} />
+
+For large leaderboards, you will need to fetch in batches of 8192 elements.
+
+<SdkExampleTabs snippetId={'API_LeaderboardFetchByRankPagination'} />
+
+### Get leaderboard length
+
+Gets the length of the leaderboard.
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md]: I might rephrase this to `Get the number of entries in the leaderboard`

File: docs/leaderboard/develop/api-reference/index.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+<SdkExampleTabs snippetId={'API_LeaderboardRemoveElements'} />
+
+For large leaderboards, you will need to remove in batches of 8192 elements.
+
+<SdkExampleTabs snippetId={'API_LeaderboardRemoveElementsPagination'} />
+
+### Delete leaderboard
+
+Deletes the leaderboard.
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/index.md]: There aren't any parameters for this one? You might explicitly state that the leaderboard identifier is inferred from the leaderboard client.

File: docs/leaderboard/develop/api-reference/language_support.md [hunk 10]
```diff
@@ -0,0 +1,11 @@
+---
+sidebar_position: 2
+sidebar_label: Language Support
+title: Language API Support Matrix
+description: A reference as to which Momento SDK languages support each API
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/language_support.md]: Let's restate this more casual/reader friendly.
`See which Momento SDKs support the Leaderboard service`

Something like that.

File: docs/leaderboard/develop/api-reference/response-objects.md [hunk 11]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (allenheltondev) [inline docs/leaderboard/develop/api-reference/response-objects.md:1]: I'm going to let it slide for now, but I think this page does not have value/is misleading. The main index page references this one stating it has specific information, but it is very generic.

File: docs/leaderboard/index.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+Go to the [Momento console](https://console.gomomento.com/tokens) and follow the instructions to log in with your email
+address, Google account, or GitHub account.
+
+![Image of Momento console landing page](@site/static/img/getting-started/console.png)
+
+Click on the Create Cache button and create a cache using the cloud provider and region you want to use for your Momento Leaderboard.
+
+![Image of Momento console Create Cache form](@site/static/img/console-create-cache-form.png)
+
+Then create an API key via the [Momento console](https://console.gomomento.com) on the Generate API Key screen by selecting the [API Keys](https://console.gomomento.com/tokens) menu option.
```
Comment (allenheltondev) [inline docs/leaderboard/index.md]: Instead of typing out the steps, we should link to the auth page that shows how to create an API key.

File: docs/leaderboard/index.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+See the [API reference](./develop/api-reference/index.md) page for documentation on each available leaderboard API.
+
+Check out the [Node.js example](https://github.com/momentohq/client-sdk-javascript/blob/main/examples/nodejs/cache/leaderboard.ts) for a runnable program that includes examples of how to call each API.
+
+## Frequently asked questions about Momento Topics
+
+<details>
+  <summary>What's the difference between Momento Leaderboard and Redis Sorted Sets?</summary>
+  Momento Leaderboard has a much more efficient memory footprint, allowing us to scale to tens of millions of records. 
+  Additionally, we will soon offer more purpose-built APIs for leaderboards, like competition ranking.
```
Comment (allenheltondev) [inline docs/leaderboard/index.md]: Is competition ranking on the schedule? I hesitate to put that in here if it's not on the active roadmap.

**Review by allenheltondev — CHANGES_REQUESTED** [review]
Do we have a pricing structure for this yet? If so, we also need to update the pricing page.

File: docs/leaderboard/develop/api-reference/index.md (hunk 3 above)
Comment (anitarua) [reply docs/leaderboard/develop/api-reference/index.md:37]: No, the [protos](https://github.com/momentohq/client-protos/blob/main/proto/leaderboard.proto#L61) are expecting a `number` for the unique identifier

File: docs/leaderboard/develop/api-reference/index.md (hunk 5 above)
Comment (anitarua) [reply docs/leaderboard/develop/api-reference/index.md:72]: No, the user will have to request more elements until they receive an empty list indicating the end of the results. I can specifically mention that here.

File: docs/leaderboard/develop/api-reference/response-objects.md (hunk 11 above)
Comment (anitarua) [reply docs/leaderboard/develop/api-reference/response-objects.md:1]: Yeah I copied this over from the other docs sections, but I agree that it's very generic right now. I can remove this page and/or the references to it if that seems like the better move

File: docs/leaderboard/index.md (hunk 13 above)
Comment (anitarua) [reply docs/leaderboard/index.md]: Got confirmation it's been deprioritized, I'll remove this note

**Comment by anitarua** [comment]
@allenheltondev there's no pricing information yet, I figured we can add that page later

File: docs/leaderboard/index.md (hunk 2 above)
Comment (pete-naylor) [reply docs/leaderboard/index.md:23]: Why are we publicly documenting (with caveats) something that will be available to exactly one customer?

**Review by allenheltondev — APPROVED** [review]
